### DISCUSSION TITLE
pagespeed warning:  Reduce unused CSS

### DISCUSSION THREAD

**Discussion by bonesoul** [vercel/next.js#40714] [Help]
So pagespeed reports this next.js specific warning:

![image](https://user-images.githubusercontent.com/73215/191245839-dc608fac-e859-4593-a7ce-94609e3674e0.png)

Although i've search examples I couldn't find a next.js example that uses tailwind, postcss, and postcss-purgecss together. I just saw this test: https://github.com/vercel/next.js/blob/0af3b526408bae26d6b3f8cab75c4229998bf7cb/test/integration/css-fixtures/with-tailwindcss-and-purgecss/postcss.config.js

I came up with this postcss.config.js though:

```
module.exports = {
  plugins: [
    'tailwindcss',
    'autoprefixer',
    'postcss-import',
    "postcss-flexbugs-fixes", [
      "postcss-preset-env",
      {
        "autoprefixer": {
          "flexbox": "no-2009"
        },
        "stage": 3,
        "features": {
          "custom-properties": false
        }
      }
    ],
    process.env.NODE_ENV === 'production'
      ? [
          '@fullhuman/postcss-purgecss',
          {
            content: [
              './pages/**/*.{js,jsx,ts,tsx}',
              './components/**/*.{js,jsx,ts,tsx}',
            ],
            defaultExtractor: (content) =>
              content.match(/[\w-/:]+(?<!:)/g) || [],
          },
        ]
      : undefined,
    'postcss-preset-env',
  ],
};
```

Any ideas how to fix / improve this warning?

**Comment by icyJoseph** [comment]
Is it fair to say that the issue here is that, because Tailwind compiles to one gigantic CSS file, **_albeit purged_**, it is possible that some pages, like the one you are measuring, do not use all of that gigantic CSS file?  How many CSS files have you got?

<img width="919" alt="Screenshot 2022-09-20 at 13 52 58" src="https://user-images.githubusercontent.com/21013447/191250690-11e774e2-cddd-4645-bbfd-6fef98bd3ad2.png">

For example look at the above, I used the Chrome Coverage Tool, and you can see that for the current page about 46% of the CSS is unused, but if I navigate to other pages, the unused CSS may go down as far as 20%. 

Could you do a similar analysis for your page? https://developer.chrome.com/docs/devtools/coverage/

## Edit

Not bashing Tailwind or anything, but it is the case that having a single CSS entry point shared between multiple pages implies that, if the pages styles are dramatically different, then each page will see more unused CSS.

**Comment by bonesoul** [comment]
Seems %81 is unused. 

![image](https://user-images.githubusercontent.com/73215/191252217-fc256a39-1dfd-414b-b5bc-e34e1488fc75.png)

I'll try moving unused stuff around.
